design/src/docs/anatomy.md

```diff
@@ -0,0 +1,687 @@
+    >
+    where
+        H: Handler<GetPokemonSpecies>,
+    {
+        self.get_pokemon_species_operation(GetPokemonSpecies::from_handler(value))
+    }
+```
+
+To finalize the build and construct the complete service, `PokemonService`, each builder has a `build` method whose constraints list out all the requirements for composition:
+
+```rust
+    /// Constructs a [`PokemonService`] from the arguments provided to the builder.
+    pub fn build(self) -> PokemonService<Route>
+    where
+        Op1: Upgradable<AwsRestJson1, CheckHealth>,
+        Op1::Service: tower::Service<http::Request, Error = Infallible>,
+
+        Op2: Upgradable<AwsRestJson1, DoNothing>,
+        Op2::Service: tower::Service<http::Request, Error = Infallible>,
+
+        /* ... */
+
+        Op6: Upgradable<AwsRestJson1, GetStorage>,
+        Op6::Service: tower::Service<http::Request, Error = Infallible>,
+```
+
+Notice the general form: `Op{N}` must upgrade to a HTTP service.
+
+We provide two builder constructors:
+
+```rust
+    /// Constructs a builder for [`PokemonService`].
+    pub fn builder() -> PokemonServiceBuilder<
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+    > {
+        PokemonServiceBuilder {
+            check_health: MissingOperation,
+            do_nothing: MissingOperation,
+            get_pokemon_species: MissingOperation,
+            get_server_statistics: MissingOperation,
+            capture_pokemon: MissingOperation,
+            get_storage: MissingOperation,
+        }
+    }
+
+    /// Constructs an unchecked builder for [`PokemonService`].
+    ///
+    /// This will not enforce that all operations are set, however if an unset operation is used at runtime
+    /// it will return status code 500 and log an error.
+    pub fn unchecked_builder() -> PokemonServiceBuilder<
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+    > {
+        PokemonServiceBuilder {
+            check_health: FailOnMissingOperation,
+            do_nothing: FailOnMissingOperation,
+            get_pokemon_species: FailOnMissingOperation,
+            get_server_statistics: FailOnMissingOperation,
+            capture_pokemon: FailOnMissingOperation,
+            get_storage: FailOnMissingOperation,
+        }
+    }
+```
+
+The `builder` constructor provides a `PokemonServiceBuilder` where `build` cannot be called until all operations are set because `MissingOperation` purposefully doesn't implement `Upgradable`. In contrast, the `unchecked_builder` which sets all `Op{N}` to `FailOnMissingOperation` can be immediately built, however any unset operations are upgraded into a service which always returns status code 500, as noted in [Upgrading a Model Service](#upgrading-a-model-service).
```

Comment:
I'd omit `unchecked_builder` entirely. It's just going to make the reader raise their eyebrows as to its utility if we don't mention here the multi-Lambda based set up use case, which is not relevant to this text.

Reply:
I'm on the fence here - if we don't have `unchecked_builder` and `FailOnMissingOperation` then we don't have an example of an alternative `Upgradable::upgrade` path which weakens the motivation for the `Upgradable` trait entirely - which might raise eyebrows.